writer.go

```diff
@@ -364,6 +384,11 @@ func diffp(new []int, old []int) (diff []int) {
 	return
 }
 
+type partitionWriter interface {
+	Messages() chan<- writerMessage
+	Close()
```

Comment:
There is no need to export those methods, they are only used internally within the package. To make the intention clear I would lowercase them.